automap/src/comm_layer/pcp_pmp_common/windows_specific.rs

```diff
@@ -0,0 +1,401 @@
+// Copyright (c) 2019-2021, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+#![cfg(target_os = "windows")]
+
+use crate::comm_layer::pcp_pmp_common::FindRoutersCommand;
+use crate::comm_layer::AutomapError;
+use std::net::IpAddr;
+use std::str::FromStr;
+
+pub fn windows_find_routers(command: &dyn FindRoutersCommand) -> Result<Vec<IpAddr>, AutomapError> {
+    match command.execute() {
+        Ok(stdout) => {
+            // Arrange to split output into line pairs that we can zip together
+            let mut firsts = stdout.split(&['\n', '\r'][..]).collect::<Vec<&str>>();
+            firsts.push("");
+            let mut seconds = vec![""];
+            seconds.extend(firsts.clone());
+            let addresses = firsts
+                .into_iter()
+                .zip(seconds)
+                .filter(|(_, first)| first.contains("Default Gateway"))
+                .map(|(second, first)| {
+                    let first_line_strs = first
+                        .split(' ')
+                        .filter(|s| s.len() >= 2)
+                        .collect::<Vec<&str>>();
+                    let second_addr_opt = match IpAddr::from_str(second.trim()) {
+                        Ok(addr) => Some(addr),
+                        Err(_) => None,
+                    };
+                    (first_line_strs, second_addr_opt)
+                })
+                .filter(|(first_line_strs, _)| first_line_strs.len() > 2)
+                .map(
+                    |(first_elements, ip_addr_opt)| match (first_elements, ip_addr_opt) {
+                        (_, Some(IpAddr::V4(ipv4_addr))) => Some(IpAddr::V4(ipv4_addr)),
+                        (first_elements, _) => {
+                            let ip_addr_maybe_with_scope_id = first_elements[2];
+                            let ip_addr_str =
+                                ip_addr_maybe_with_scope_id.split('%').collect::<Vec<_>>()[0];
+                            match IpAddr::from_str(ip_addr_str) {
+                                Err(_) => panic!("Bad syntax from ipconfig /all"),
+                                Ok(addr) => Some(addr),
```

Comment:
This could be something like:
```
IpAddr::from_str(ip_addr_str).map_err(|_|panic!("Bad syntax from ipconfig /all")).ok()
```
But I don't insist on that...